File: zhtj-ui/src/api/activity.ts
import api from './index';

export interface Activity {
  id?: number;
  organization?: number;
  category: string;
  requiredTopic?: string;
  optionalTopics?: string;
  participants?: string;
  date: string;
  place: string;
  content: string;
  host?: string;
  organizationName?: string;
  createTime?: string;
  updateTime?: string;
}

// 添加活动
export const addActivity = (data: Activity): Promise<any> => {
  return api.post('/api/activities', data);
};

// 删除活动
export const deleteActivity = (id: number): Promise<any> => {
  return api.delete(`/api/activities/${id}`);
};

// 修改活动
export const updateActivity = (data: Activity): Promise<any> => {
  if (!data.id) {
    return Promise.reject('活动ID不能为空');
  }
  return api.put(`/api/activities/${data.id}`, data);
};

// 获取组织的所有活动
export const getActivities = (params?: { 
  page?: number; 
  size?: number; 
  organization?: number;
  category?: string;
  startDate?: string;
  endDate?: string;
}): Promise<any> => {
  return api.get('/api/activities', { params });
};

// 获取活动详情
export const getActivityDetail = (id: number): Promise<any> => {
  return api.get(`/api/activities/${id}`);
};

// 根据日期查询活动
export const getActivitiesByDate = (params: { 
  organization: number; 
  date: string 
}): Promise<any> => {
  return api.get('/api/activities/date', { params });
};

// 获取活动类别统计
export const getActivityStatistics = (organization: number): Promise<any> => {
  return api.get('/api/activities/statistics/category', { 
    params: { organization } 
  });
}; 

File: zhtj-springboot/src/main/java/com/zhtj/domain/Activity.java
package com.zhtj.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("activity")
public class Activity {
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;             // 序号
    
    private Integer organization;   // 团组织序号
    
    private String category;        // 类别(支部大会/团课/主题团日/入团仪式/组织生活会/其他)
    
    private String requiredTopic;   // 必学专题
    
    private String optionalTopics;  // 自学专题(多选,逗号分隔)
    
    private String participants;    // 参加的团员
    
    private String date;            // 开展时间
    
    private String place;           // 地点
    
    private String content;         // 内容
    
    private String host;            // 活动主持人
    
    private LocalDateTime createTime; // 创建时间
    
    private LocalDateTime updateTime; // 更新时间
    
    @TableField(exist = false)
    private String organizationName; // 组织名称，非数据库字段
} 